cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -295,9 +295,7 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 		klog.V(1).Info("No unschedulable pods")
 	} else if a.MaxNodesTotal > 0 && len(readyNodes) >= a.MaxNodesTotal {
 		scaleUpStatus.Result = status.ScaleUpNoOptionsAvailable
```

Comment:
Let the `Result` also be filled by `ScaleUp()` and totally omit this `else if`. WDYT?

Reply:
I don't really want to change behaviour. All I want is the event emitted when the cluster can scale up no more. I'm hoping that's the smaller, simplest and most discrete change.

Reply:
I am fine if you want to take care of event part only. Though i think it wont change behavior if we move the filling up `Result` to the ScaleUp, leave it to you to decide whether to include in this PR or not.